src/twisted/trial/test/test_loader.py

```diff
@@ -98,33 +98,26 @@ def test_moduleNotInPath(self):
         L{runner.filenameToModule} returns a module object loosely
         resembling the module defined by that file anyway.
         """
-        # "test_sample" isn't actually the name of this module.  However,
-        # filenameToModule can't seem to figure that out.  So clean up this
-        # misnamed module.  It would be better if this weren't necessary
-        # and filenameToModule either didn't exist or added a correctly
-        # named module to sys.modules.
-        self.addCleanup(sys.modules.pop, "test_sample", None)
 
         self.mangleSysPath(self.oldPath)
         sample1 = runner.filenameToModule(
             os.path.join(self.parent, "goodpackage", "test_sample.py")
         )
+        self.cleanUpModules()
         self.mangleSysPath(self.newPath)
         from goodpackage import test_sample as sample2  # type: ignore[import]
 
-        self.assertEqual(
-            os.path.splitext(sample2.__file__)[0], os.path.splitext(sample1.__file__)[0]
-        )
+        self.assertIsNot(sample1, sample2)
+        self.assertEqual(sample1.__spec__, sample2.__spec__)
 
     def test_packageInPath(self):
         """
         If the file in question is a package on the Python path, then it should
         properly import and return that package.
         """
         package1 = runner.filenameToModule(os.path.join(self.parent, "goodpackage"))
-        import goodpackage
 
-        self.assertEqual(goodpackage, package1)
+        self.assertIs(package1, sys.modules["goodpackage"])
```

Comment:
part of "properly import and return that package." is adding the package to `sys.modules`

Reply:
but isn't `import goodpackage` doing the same thing?
just asking :)

Reply:
It will set the sys.modules key if we forgot to set it